File: 07_funcoes/ex01_calculadora.c
/*
Exercicio 01
Tema: 07_funcoes
*/

#include <stdio.h>

void somar(int n1, int n2, char op){
    printf("%d + %d = %d\n", n1, n2, n1+n2);
}
void subtrair(int n1, int n2, char op){
    printf("%d - %d = %d\n", n1, n2, n1-n2);
}
void multiplicar(int n1, int n2, char op){
    printf("%d * %d = %d\n", n1, n2, n1*n2);
}
void dividir(int n1, int n2, char op){
    if(n2 == 0){
        printf("Nao se pode dividir por '0'\n");
        return;
    }
    printf("%d / %d = %d\n", n1, n2, n1/n2);
}   

int main() {

    int num1, num2;
    char op;

    printf("Digite o primeiro numero: ");
    scanf("%d", &num1);
    printf("Digite a operacao: ");
    scanf(" %c", &op);
    printf("Digite o segundo numero: ");
    scanf("%d", &num2);

    switch(op){
        case '+':
            somar(num1,num2,op);
            break;

            case '-':
            subtrair(num1,num2,op);
            break;

        case '*':
            multiplicar(num1,num2,op);
            break;

        case '/':
            dividir(num1,num2,op);
            break;
        
        default:
        printf("Operacao invalida!");
    }

    return 0;
}

File: 07_funcoes/ex01_calculadora.py
"""
Exercicio 01
Tema: 07_funcoes
"""

def somar(n1, n2, op):
    print(f"{n1} + {n2} = {n1+n2}")
def subtrair(n1, n2, op):
    print(f"{n1} - {n2} = {n1-n2}")
def multiplicar(n1, n2, op):
    print(f"{n1} * {n2} = {n1*n2}")
def dividir(n1, n2, op):
    if n2 == 0:
        print("Nao pode dividir numero por '0'")
        return
    print(f"{n1} / {n2} = {n1/n2}")

n1 = int(input("Digite o primeiro numero: "))
op = input("Digite a operacao: ")
n2 = int(input("Digite o segundo numero: "))

match op:
    case '+':
        somar(n1,n2,op)
        
    case '-':
        subtrair(n1,n2,op)
        
    case '*':
        multiplicar(n1,n2,op)
        
    case '/':
        dividir(n1,n2,op)
        
    case _:
        print("Operacao invalida!")
